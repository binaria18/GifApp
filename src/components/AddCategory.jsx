import { useState } from 'react'
import React from 'react';

export const AddCategory = ( { setCategories, categories } ) => {

    const [inputValue, setInputValue] = useState('')

    const adCategory = () => {
        if(inputValue.trim().length <= 1) return
        if(categories?.includes(inputValue)) return
        setCategories([inputValue])
    }

    const limpiarFormulario = () => {
        setInputValue('')
        document.getElementById('category').value = ''
    }
    const onsubmit = (e) => {
        e.preventDefault()
        adCategory()    
        setInputValue('')
        document.getElementById('category').value = ''
    }
    return (
        <>
            <form onSubmit={ (e) => onsubmit(e) } aria-label='form'>
                <input 
                    onChange={(e) => setInputValue(e.target.value)} 
                    placeholder='Nueva categoría'
                    type="text" 
                    id="category"/>
                <button type="submit" id="btnAddCategory">Agregar</button>
                <button onClick={limpiarFormulario} id="btnLimpiar">Limpiar</button>
            </form>
        </>
    )
}