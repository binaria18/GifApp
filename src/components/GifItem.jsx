import React from 'react'

export const GifItem = ({ gif }) => { 

    return (
        <div className="gifCard">
            <img 
                key={ gif.id } src={ gif.url } alt={ gif.title } 
                className="gifCard" />
            <div className='gifText'>
                { gif.title }
            </div>
        </div>
    )
}