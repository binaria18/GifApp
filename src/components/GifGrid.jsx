import React from 'react'
import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'


/* Componente, recibe una categoría por parámetro y devuelve los gifs */
export const GifGrid = ({ category }) => { 

    const { misGifs, isLoading } = useFetchGifs( category ) 
    
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading && <h2>Cargando...</h2>
            }
            <div className='card-grid'>
                {/* Mapeo de gifs */}
                {   
                    misGifs.map( gif => 
                        <GifItem key={ gif.id } gif={ gif } />
                    ) 
                    
                }
            </div>
        </>
    )
}

export default GifGrid
