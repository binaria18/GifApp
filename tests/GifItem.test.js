/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { GifItem } from '../src/components/GifItem';

describe('Test GifGrid', () => {
    it('Debe mostrar la imagen con el URL y el ALT indicados', () => {
        render(<GifItem gif={{ id: '1', title: 'Test Title', url: 'https://example.com/test.gif' }} />);
        const img = screen.getByRole('img');
        expect(img.src).toBe('https://example.com/test.gif');
        expect(img.alt).toBe('Test Title');
        //screen.debug();
    })
    it('Debe mostrar el título', () => {
        render(<GifItem gif={{ id: '1', title: 'Test Title', url: 'https://example.com/test.gif' }} />);
        const title = screen.getByText('Test Title');
        expect(title).toBeTruthy();
    })
    it('Debe hacer match con el Snapshot', () => {
        render(<GifItem gif={{ id: '1', title: 'Test Title', url: 'https://example.com/test.gif' }} />);
        const { container } = render(<GifItem gif={{ id: '1', title: 'Test Title', url: 'https://example.com/test.gif' }} />);
        expect(container).toMatchSnapshot(); // El snapshot se crea en la primera ejecución
    })
});