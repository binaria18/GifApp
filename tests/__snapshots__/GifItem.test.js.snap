// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="gifCard"
  >
    <img
      alt="Test Title"
      class="gifCard"
      src="https://example.com/test.gif"
    />
    <div
      class="gifText"
    >
      Test Title
    </div>
  </div>
</div>
`;

exports[`Test gifItem Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="gifCard"
  >
    <img
      alt="Test Title"
      class="gifCard"
      src="https://example.com/test.gif"
    />
    <div
      class="gifText"
    >
      Test Title
    </div>
  </div>
</div>
`;
